src/SmartTransactionsController.ts

```diff
@@ -333,42 +398,58 @@ export default class SmartTransactionsController extends BaseController<
     });
   }
 
-  async updateSmartTransactions() {
+  async updateSmartTransactions({
+    networkClientId,
+  }: {
+    networkClientId?: NetworkClientId;
+  } = {}): Promise<void> {
     const { smartTransactions } = this.state.smartTransactionsState;
-    const { chainId } = this.config;
+    const chainId = this.getChainId({ networkClientId });
+    const smartTransactionsForChainId = smartTransactions?.[chainId];
```

Comment:
```suggestion
    const smartTransactionsForChainId = smartTransactions[chainId];
```

This can be removed without causing errors.